hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -235,6 +235,29 @@ protected void initTableMetadata() {
   protected void bootstrapIfNeeded(HoodieEngineContext engineContext, HoodieTableMetaClient datasetMetaClient) throws IOException {
     HoodieTimer timer = new HoodieTimer().startTimer();
     boolean exists = datasetMetaClient.getFs().exists(new Path(metadataWriteConfig.getBasePath(), HoodieTableMetaClient.METAFOLDER_NAME));
+    boolean rebootstrap = false;
+    if (exists) {
+      // If the un-synched instants have been archived then the metadata table will need to be bootstrapped again
```

Comment:
@prashantwason Thinking about this after today's discussion, will there ever be a situation when un-synced instants will be archived ? Given the `syncMetadataTable` happens before everything (in the AbstractHoodieWriteClient) and after commit, there should never be an instance like this ?

Reply:
This can happen if the metadata table is disabled (say while debugging production issues) and later enabled.